### ISSUE TITLE
As-built Report hangs with no errors.

### ISSUE THREAD

**Issue by keekeh-007** [AsBuiltReport/AsBuiltReport.Microsoft.AD#161]
### Bug description

Process stuck for me at the AsBuiltReport.Microsoft.AD 0.8.0 is currently installed line
![Asbuiltstuck](https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD/assets/167115851/1038b56e-9404-4393-9e60-4bc10bf4cfae)

### Command-line input
$Creds = Get-Credential
New-ASBuiltReport -Report Microsoft.AD -Target server01.contoso.com -Credential $Creds -Format Html,Text -OutputFolderPath 'C:\Temp\' -EnableHealthCheck -AsBuiltConfigFilePath "C:\Users\test\AsBuiltReport\AsBuiltReport.json" -ReportConfigFilePath "C:\Users\test\AsBuiltReport\AsBuiltReport.Microsoft.AD.json"

### Steps to reproduce

I've tried running this same command line from both AD server and Windows 10 machine.

### Expected behaviour

Report to be completed.

### Operating System

Tried from Windows 10 and Server 2016

### PowerShell Version

5.1.14393.6343

### PowerShell Modules
AsBuiltReport.Core              1.4.0
AsbuiltReport.Microsoft.AD 0.8.0
PScribo                                 0.10.0.
PScriboCharts                      0.9.0
PSPKI                                   4.2.0

### Additional Context

This used to work for me in the past but now it just seems to hang or never finish.  Please let me know if I can do anything to remediate from my end.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://www.asbuiltreport.com/about/contributing/#reporting-issues-and-bugs).
- [X] I have read [the documentation](https://www.asbuiltreport.com/user-guide/new-asbuiltconfig), and referred to the [known issues](https://www.asbuiltreport.com/user-guide/known-issues/) before submitting this bug report.
- [X] I have checked for previously opened & closed [issues](https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD/issues) before submitting this bug report.

**Comment by rebelinux** [comment]
Can you run the report with the Verbose option and see if any errors message  is displayed?

**Comment by keekeh-007** [comment]
I started it with just -Verbose after the initial command line.  Let me try again changing the global verbose parameters as well.  Do I also include the -Verbose switch?

**Comment by rebelinux** [comment]
> I started it with just -Verbose after the initial command line. Let me try again changing the global verbose parameters as well. Do I also include the -Verbose switch?

Yes.

**Comment by rebelinux** [comment]
That section looks for missing subnet in the netlogon.log file of each DC maybe a slow bandwidth is slowing the connection. 

Anyway 14 minutes is a long time I am not sure why the cmdlet is not getting a timeout event. 

```powershell
if (Test-Connection -ComputerName $DC -Quiet -Count 2) {
    try {
        $DCPssSession = New-PSSession $DC -Credential $Credential -Authentication $Options.PSDefaultAuthentication -Name 'MissingSubnetinAD'
        $Path = "\\$DC\admin`$\debug\netlogon.log"
        if ((Invoke-Command -Session $DCPssSession { Test-Path -Path $using:path }) -and (Invoke-Command -Session $DCPssSession { (Get-Content -Path $using:path | Measure-Object -Line).lines -gt 0 })) {
            Write-PScriboMessage "Collecting Missing Subnet in AD information from $($Domain)."
            $NetLogonContents = Invoke-Command -Session $DCPssSession { (Get-Content -Path $using:Path)[-200..-1] }
            foreach ($Line in $NetLogonContents) {
                if ($Line -match "NO_CLIENT_SITE") {
                    $inObj = [ordered] @{
                        'DC' = $DC
                        'IP' = $Line.Split(":")[4].trim(" ").Split(" ")[1]
                    }

                    $OutObj += [pscustomobject]$inobj
                }

                if ($HealthCheck.Site.BestPractice) {
                    $OutObj | Set-Style -Style Warning -Property 'IP'
                }
            }
        } else {
            Write-PScriboMessage "Unable to read $Path on $DC"
        }
        if ($DCPssSession) {
            Remove-PSSession -Session $DCPssSession
        }
    } catch {
        Write-PScriboMessage -IsWarning "Missing Subnet in AD Item Section: $($_.Exception.Message)"
    }
}
```

**Comment by rebelinux** [comment]
Can you add the that DC to the Exclude.DCs	option so we can rule out if it is a DC connection problem?

**Comment by rebelinux** [comment]
Sorry forgot to add the image:

The Exclude.DCs is located at the AsBuiltReport.Microsoft.AD.json

You can generate the file using this docs link:
https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD?tab=readme-ov-file#pencil2-configuration

![image](https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD/assets/1002783/76ced1b6-0b18-4c51-87f1-799a8bffb49f)

**Comment by keekeh-007** [comment]
> Can you add the that DC to the Exclude.DCs option so we can rule out if it is a DC connection problem?

I executed repadmin /showrepl and found a problem replicating to a DR.

> Sorry forgot to add the image:
> 
> The Exclude.DCs is located at the AsBuiltReport.Microsoft.AD.json
> 
> You can generate the file using this docs link: https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD?tab=readme-ov-file#pencil2-configuration
> 
> ![image](https://private-user-images.githubusercontent.com/1002783/322573222-76ced1b6-0b18-4c51-87f1-799a8bffb49f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMDI1NzMsIm5iZiI6MTcxMzIwMjI3MywicGF0aCI6Ii8xMDAyNzgzLzMyMjU3MzIyMi03NmNlZDFiNi0wYjE4LTRjNTEtODdmMS03OTlhOGJmZmI0OWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTVUMTczMTEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmVmYzI2ZjdjZjI2ODg3OGQ3OTcyOWI4YmI5MmQ0ODVlMjM0MDllZDIwOGVjZjZhNzEwNzU5MmVhOGY3YWI4MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PKdstUlEujKVCTze3_aXV_YG1NEK60S_MGseuDZEtwo)

I had gotten, Invalid JSON primitive: dc01.contoso.com (I think I was missing the quotes on the json file around the server name)

**Comment by rebelinux** [comment]
Yes, the quotes are a requirement in json

**Comment by keekeh-007** [comment]
> Yes, the quotes are a requirement in json

Thank you, it's further along.  Update to follow: 
![image](https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD/assets/167115851/ea139b39-c72c-4e10-b395-866e6151d95c)

**Comment by keekeh-007** [comment]
Now it seems to be stuck in this process: (about 15 minutes now)
![image](https://github.com/AsBuiltReport/AsBuiltReport.Microsoft.AD/assets/167115851/11d646dd-56ef-4b57-b9da-94a3dabc9c58)

**Comment by keekeh-007** [comment]
> If still stuck in that section you can try to disable the $HealthCheck.Domain.DFS in the AsBuiltReport.Microsoft.AD.json file.
> 
> Set the value to false to disable it and rerun the report.
> 
> ![image](https://private-user-images.githubusercontent.com/1002783/322591144-4ec7b876-aa09-4ece-8690-fc08be3d8b8b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMDY3NTEsIm5iZiI6MTcxMzIwNjQ1MSwicGF0aCI6Ii8xMDAyNzgzLzMyMjU5MTE0NC00ZWM3Yjg3Ni1hYTA5LTRlY2UtODY5MC1mYzA4YmUzZDhiOGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTVUMTg0MDUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDk4Y2I3ZTFmNWRhYWI0ZTc5ZGJmZGM2OThhNTdiMDU3MDE5YmU1OGViODQxNzYwZmYzZjY4MjQ3OTdkNTY1ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6m2URalJelJFuvhRe9OBCJla9X_DW39ePmw7wYcbi9Y)
> 
> However, I think there must be a connection problem in your environment.

Thank you for your help,  I will attempt this workaround step.  Is there a way to set timeouts to continue throug the reporting process?

**Comment by rebelinux** [comment]
Currently there is no way to set a timeout in the report.

Powershell have the capability to set a timeout for a job but coding it for a multi-Domain/DC is very difficult. Also there is no right way to define a timeout for a cmdlet that is called remotely in a distributed Active Directory environment.

There are also many thing to take into consideration like bandwidth, server resources and implemented security mechanism in the environment that can prevent the report to run successfully.

Maybe in a future release a per job timeout can be implemented.

**Comment by keekeh-007** [comment]
> Currently there is no way to set a timeout in the report.
> 
> Powershell have the capability to set a timeout for a job but coding it for a multi-Domain/DC is very difficult. Also there is no right way to define a timeout for a cmdlet that is called remotely in a distributed Active Directory environment.
> 
> There are also many thing to take into consideration like bandwidth, server resources and implemented security mechanism in the environment that can prevent the report to run successfully.
> 
> Maybe in a future release a per job timeout can be implemented.

It looks like after resolving the memory allocation issues with one of the DC and replication flowed across the environment the reports is now flowing right through.  Thanks again for all your time and help!

**Comment by rebelinux** [comment]
If you manage to generate the report. Could you evaluate the content and give me recommendations?

I am open to new ideas or sections that may be needed.

Best regards,
